docs/sent-l2-confirmation/sent-confirmations.md

```diff
@@ -0,0 +1,357 @@
+# SENT (L2 chain) -> Oxen chain interaction
+
+Oxen 11, as has been widely discussed for some time, moves the fundamental registration, staking,
+and rewards to an Ethereum L2 chain (Arbitrum), but still persists the Oxen chain at the service
+node layer for two main purposes:
+
+- retaining support for Oxen currency transactions.  Although we expect the majority of OXEN to be
+  converted to SENT during the Session network upgrade, we will retain support for OXEN transactions
+  for the near future to allow held OXEN to still be swapped for SENT.
+
+- Using the oxen chain as the back end "state chain" for the oxen service node network.  This allows
+  Oxen nodes to manage earned rewards amounts, to self-police the network (decomms, recomms,
+  deregs), and to precisely control when information from the L2 network starts applying to the Oxen
+  chain without needing to add wallets (with live funds) to Service Nodes.
+
+The complication of such a mechanism is that, unlike when dealing with just one chain, the
+interactions can get complicated in various ways:
+
+- An L2 RPC provider might be lagging behind, in which case the  registrataions and other activity
+  that it learns about would lag the rest of the Oxen network, causing it to fall behind.
+
+- A malicious L2 provider who provides a significant amount of the Oxen network could provide false
+  information regarding registration activity.
```

Comment:
A L2 RPC provider could inject fake transactions and get it relayed across to all the other nodes. The other nodes will then provide confirmation on these transactions causing havoc when they weren't actually in the L2. 

This seems like the most feasible attack. The P2P code is pretty complicated, I saw that when we add a L2 transaction we mark it as do-not-relay. Is there a distinction in the P2P layer from TXs that are sent via relay with zero-confirmations and TXs that are sent as part of syncing the block (e.g. somewhat confirmed)? 

The idea being we reject L2 TXs on the P2P layer if they aren't in a block because we're only meant to observe these TXs from L2 whilst our RPC node still has the data. If we go out of the available range of blocks in the RPC provider then we just have to trust the node you're syncing from.

Reply:
> This seems like the most feasible attack. The P2P code is pretty complicated, I saw that when we add a L2 transaction we mark it as do-not-relay. Is there a distinction in the P2P layer from TXs that are sent via relay with zero-confirmations and TXs that are sent as part of syncing the block (e.g. somewhat confirmed)?

The do-not-relay code is not actually necessary anymore because when the pulse quorum signs off on a transaction, it requires that any L2 state changes are known in the L2 tracker, not the mempool (and likewise for subsequent pulse quora casting confirmation votes).

Reply:
Confirmed not a problem. This is the code in `create_block_template_internal`

```
b.l2_votes = service_node_list.l2_pending_state_votes();
```

Which applies votes based on what's in the L2 tracker, _not_ the tx pool